ecs-cli/modules/cli/cluster/cluster_app_test.go

```diff
@@ -961,6 +962,73 @@ func TestClusterUpWithEmptyClusterWithExistingStack(t *testing.T) {
 	assert.Error(t, err, "Unexpected error bringing up empty cluster")
 }
 
+func TestClusterUpARM64(t *testing.T) {
+	defer os.Clearenv()
+	mockECS, mockCloudformation, mockSSM := setupTest(t)
+	awsClients := &AWSClients{mockECS, mockCloudformation, mockSSM}
+
+	gomock.InOrder(
+		mockECS.EXPECT().CreateCluster(clusterName).Return(clusterName, nil),
+	)
+
+	gomock.InOrder(
+		mockSSM.EXPECT().GetRecommendedECSLinuxAMI("arm64").Return(amiMetadata(armAMIID), nil),
+	)
+
+	gomock.InOrder(
+		mockCloudformation.EXPECT().ValidateStackExists(stackName).Return(errors.New("error")),
+		mockCloudformation.EXPECT().CreateStack(gomock.Any(), stackName, true, gomock.Any()).Do(func(x, y, w, z interface{}) {
+			capabilityIAM := w.(bool)
+			cfnParams := z.(*cloudformation.CfnStackParams)
+			amiIDParam, err := cfnParams.GetParameter(ParameterKeyAmiId)
+			assert.NoError(t, err, "Unexpected error getting cfn parameter")
+			assert.Equal(t, armAMIID, aws.StringValue(amiIDParam.ParameterValue), "Expected ami ID to be set to recommended for arm64")
+			assert.True(t, capabilityIAM, "Expected capability capabilityIAM to be true")
+		}).Return("", nil),
+		mockCloudformation.EXPECT().WaitUntilCreateComplete(stackName).Return(nil),
+	)
+
+	flagSet := flag.NewFlagSet("ecs-cli-up", 0)
+	flagSet.Bool(flags.CapabilityIAMFlag, true, "")
+	flagSet.String(flags.KeypairNameFlag, "default", "")
+	flagSet.String(flags.InstanceTypeFlag, "a1.medium", "")
+
+	context := cli.NewContext(nil, flagSet, nil)
+	rdwr := newMockReadWriter()
+	commandConfig, err := newCommandConfig(context, rdwr)
+	assert.NoError(t, err, "Unexpected error creating CommandConfig")
+
+	err = createCluster(context, awsClients, commandConfig)
+	assert.NoError(t, err, "Unexpected error bringing up cluster")
+}
+
+func TestDetermineArchitecture(t *testing.T) {
+	var testCases = []struct {
+		in  string
+		out string
+	}{
+		{"a1.medium", "arm64"},
+		{"a1.large", "arm64"},
+		{"a1.xlarge", "arm64"},
+		{"a1.2xlarge", "arm64"},
+		{"a1.4xlarge", "arm64"},
+		{"t2.medium", "x86"},
+		{"c5.large", "x86"},
+		{"i3.metal", "x86"},
+		{"t3.micro", "x86"},
+	}
```

Comment:
Should we also add an error case?

Reply:
There aren't any cases; the regex code I have doesn't have the ability to check if something is a valid instance type. It can just determine if something looks like an A1 instance type, and if it doesn't, it assumes its x86.

Reply:
I suppose I was thinking that since`determineArchitecture` returns an error in its method signature we should include error cases for thoroughness. Looks like the only other error expected would be from CFN, so I suppose it's fine.